src/js/containers/AppList.js

```diff
@@ -23,16 +25,33 @@ const mapStateToProps = (state) => {
             devicename: devicename,
             image: icon,
             link: '/' + link,
-            cmdID: app.appID
+            cmdID: app
         }
     })
     return {data: data}
 }
 
 const mapDispatchToProps = (dispatch) => {
     return {
-        onSelection: (appID) => {
-            sdlController.onAppActivated(appID)
+        onSelection: (appID, app) => {
+            let state = store.getState();
+            var webEngineApp = state.appStore.installedApps.find(x => x.policyAppID === app.policyAppID);
+
+            if (!webEngineApp) {
+                sdlController.onAppActivated(appID)
+                return;
+            }
+
+            dispatch(webEngineAppLaunch(app.policyAppID, appID));
+
+            var activateAppOnceRegistered = setInterval(() => {
+                if (!app.isRegistered) {
```

Comment:
Does this value change once it is registered? IE if it fails the first time can it succeed after?

Reply:
Yes, with setInterval this function will be called every so often until the app is registered in which case it will stop waiting and send onAppActivated.